sysbuild/CMakeLists.txt

```diff
@@ -287,8 +287,13 @@ function(${SYSBUILD_CURRENT_MODULE_NAME}_pre_cmake)
       set_config_bool(mcuboot CONFIG_BOOT_FIH_PROFILE_DEFAULT_LOW y)
     endif()
 
-    if(SB_CONFIG_PARTITION_MANAGER)
-      # Use NCS signing script with support for PM
+    set(direct_xip 0)
+    if(SB_CONFIG_MCUBOOT_MODE_DIRECT_XIP OR SB_CONFIG_MCUBOOT_MODE_DIRECT_XIP_WITH_REVERT)
+      set(direct_xip 1)
+    endif()
+
+    if(SB_CONFIG_PARTITION_MANAGER OR direct_xip)
```

Comment:
do not use a temp variable which makes reading this harder, just check the Kconfigs directly

Reply:
Ok, done, however it might be a bit problematic in the future when new features which need the image_signing.cmake script are ported (such as compression/ram loader)